clients/vscode-hlasmplugin/src/branchDecorator.ts

```diff
@@ -131,63 +131,76 @@ export function activateBranchDecorator(context: vscode.ExtensionContext, client
         const [resultPromise, cancel] = getCancellableBranchInfo(client, uri, delay);
         pendingRequests.set(documentUri, cancel);
 
-        return await resultPromise.finally(() => {
+        return resultPromise.finally(() => {
             if (pendingRequests.get(documentUri) === cancel)
                 pendingRequests.delete(documentUri);
         });
     };
+    const cancelRequest = (uri: vscode.Uri) => { pendingRequests.get(uri.toString())?.() };
 
-    const activeChanged = async (editor: vscode.TextEditor) => {
-        const document = editor.document;
-        if (document.languageId !== 'hlasm') return;
+    return { scheduleRequest, cancelRequest };
 
-        const version = document.version;
+}
 
-        const result = await scheduleRequest(document.uri, requestTimeout);
+function decorationsEnabled(document: vscode.TextDocument, expectedVersion: number | undefined = undefined) {
+    return !document.isClosed
+        && document.languageId === 'hlasm'
+        && (expectedVersion === undefined || document.version === expectedVersion)
```

Comment:
Just a mere suggestion to switch the parameters for readability sake

```suggestion
        && (expectedVersion === undefined || expectedVersion === document.version)
```